### DISCUSSION TITLE
Can  we just replace the snakeyaml-1.33.jar to snakeyaml-2.0.jar  in opa-libs in apache-skywalking-apm-bin( version 9.3.0)

### DISCUSSION THREAD

**Discussion by DeBruyne2020** [apache/skywalking#12129] [Q&A]
We dont want to upgrade our jdk verision,(which the jdk we use is jdk 8) so the highest version of skywalking-oap we can use is version  9.3.0.

We need to solve the CVE-2022-1471 about snakeyaml.To solve the CVE-2022-1471 about snakeyaml we need to upgrade the snakeyaml  version to 2.0.
 

My question is  can we just replace the snakeyaml-1.33.jar to snakeyaml-2.0.jar  in opa-libs in apache-skywalking-apm-bin( version 9.3.0) ?  

  Can skywalking work normal by this operation？

🙏

**Comment by wu-sheng** [comment]
This is hard to tell. 2.0 compatibility is unknown.

**Reply by wu-sheng** [reply]
The latest release should include fixed version.
We don't check in this, we just fixed the version and bump up in next release.

**Comment by kezhenxu94** [comment]
No. It requires code changes so you can’t simply replace the lib in the distribution tar

**Reply by kezhenxu94** [reply]
You have to upgrade skywalking altogether
